File: seminar-03/D/python/D.py
def quick_sort(a):
    if len(a) <= 1:
        return a

    pivot = a[len(a) // 2]
    left = [x for x in a if x < pivot]
    mid = [x for x in a if x == pivot]
    right = [x for x in a if x > pivot]

    return quick_sort(left) + mid + quick_sort(right)


n = int(input())
a = list(map(int, input().split()))
a = quick_sort(a)
unique = 1
for i in range(1, n):
    if a[i] != a[i - 1]:
        unique += 1
print(unique)

File: seminar-03/D/java/D.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class D {

    static List<Integer> quickSort(List<Integer> a) {
        if (a.size() <= 1) {
            return a;
        }

        int pivot = a.get(a.size() / 2);
        List<Integer> left = new ArrayList<>();
        List<Integer> mid = new ArrayList<>();
        List<Integer> right = new ArrayList<>();

        for (int x : a) {
            if (x < pivot) {
                left.add(x);
            } else if (x == pivot) {
                mid.add(x);
            } else {
                right.add(x);
            }
        }

        List<Integer> res = new ArrayList<>();
        res.addAll(quickSort(left));
        res.addAll(mid);
        res.addAll(quickSort(right));
        return res;
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        List<Integer> a = new ArrayList<>();

        if (n > 0) {
            String[] parts = br.readLine().split(" ");
            for (int i = 0; i < n; i++) {
                a.add(Integer.parseInt(parts[i]));
            }
        }

        a = quickSort(a);

        int unique = 1;
        for (int i = 1; i < n; i++) {
            if (!a.get(i).equals(a.get(i - 1))) {
                unique++;
            }
        }

        System.out.println(unique);
    }
}
